File: back_end/models/prescription.js
const mongoose = require('mongoose');

const ordonnancesSchema = new mongoose.Schema({
  patient: { type: mongoose.Schema.Types.ObjectId, ref: 'Patient', required: true },
  provider:{type: mongoose.Schema.Types.ObjectId, ref: 'User', required: false },
  médicament: { type: String, required: true, minlength: 3 },
  dosage: { type: String, required: true, enum: ['1 pill', '2 pills', '1 tablet', '2 tablets', '1 capsule', '2 capsules'] },
  fréquence: { type: String, required: true, enum: ['daily', 'twice per day', 'every 6 hours'] },
  dateDébut: {
    type: Date, required: true, validate: {
      validator: function (v) {
        return v < this.dateFin;
      },
      message: 'La date de début doit être antérieure à la date de fin.'
    }
  },
  dateFin: { type: Date, required: true },
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now },
});

ordonnancesSchema.pre('save', function (next) {
  this.updatedAt = new Date();
  next();
});

const Prescription = mongoose.model('Prescription', ordonnancesSchema);

module.exports = Prescription;

File: medilink_admin/lib/models/prescription.dart

class Prescription {
   String? id;
   String? patient;
   String? provider;
  String? medication;
  String? dosage;
  String? frequency;
  DateTime? startDate;
  DateTime? endDate;
  DateTime? createdAt;
  DateTime? updatedAt;
  DateTime? reminderTime;
  List<String>?sharedwith;

  Prescription({
    this.id,
    this.patient,
    this.provider,
    this.medication,
    this.dosage,
    this.frequency,
    this.startDate,
    this.endDate,
    this.createdAt,
    this.updatedAt,
    this.reminderTime,
    this.sharedwith
  });

  factory Prescription.fromJson(Map<String, dynamic> json) {
    List<String>? sharedwith;
    if (json['sharedwith'] != null) {
      sharedwith = [];
      json['sharedwith'].forEach((appId) {
        sharedwith!.add(appId);
      });
    }
    return Prescription(
      id: json['_id']  as String?,
      patient: json['patient']  as String?,
      provider: json['provider']  as String?,
      medication: json['médicament'] as String?,
      dosage: json['dosage'] as String?,
      frequency: json['fréquence'] as String?,
      startDate: json['dateDébut'] != null ? DateTime.parse(json['dateDébut']) : null,
      endDate: json['dateFin'] != null ? DateTime.parse(json['dateFin']) : null,
      createdAt: json['createdAt'] != null ? DateTime.parse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.parse(json['updatedAt']) : null,
      sharedwith: sharedwith,
      reminderTime: json['reminderTime'] != null ? DateTime.parse(json['reminderTime']) : null,
    );
  }
}
